src/viewers/VivViewer.js

```diff
@@ -44,9 +50,19 @@ export default class VivViewer extends PureComponent {
    * This updates the viewState as a callback to the viewport changing in DeckGL
    * (hence the need for storing viewState in state).
    */
-  _onViewStateChange({ viewId, viewState, oldViewState }) {
+  _onViewStateChange({ viewId, viewState, interactionState, oldViewState }) {
     // Save the view state and trigger rerender.
-    const { views } = this.props;
+    const { views, onViewStateChange } = this.props;
+    // eslint-disable-next-line no-param-reassign
+    viewState =
+      (onViewStateChange &&
+        onViewStateChange({
+          viewId,
+          viewState,
+          interactionState,
+          oldViewState
+        })) ||
+      viewState;
```

Comment:
Can you explain the use-case for altering the view state?  I am just noticing this on final review.  I would expect this to simply be a callback since `filterViewState` on the `VivView` class effectively does this.

Reply:
I haven't seen hooks like this that alter (some of) their arguments, hence my question.

Reply:
I currently don't have a use case for this. At the same time, I see no reason not to expose that part of `Deck.onViewStateChange`: https://deck.gl/docs/api-reference/core/deck#onviewstatechange

Reply:
Ok, yes this is what that function does so it's ok to hoist that up to allow for customisation.  Can you add the link to deck.gl docs at the PIP and SBS viewer docs?

Reply:
I copied the parameter documentation from `VivViewer` that includes the link.